packages/web3-wrapper/src/web3_wrapper.ts

```diff
@@ -377,6 +381,28 @@ export class Web3Wrapper {
         const txReceipt = await txReceiptPromise;
         return txReceipt;
     }
+    /**
+     * Waits for a transaction to be mined and returns the transaction receipt.
+     * Unlike awaitTransactionMinedAsync, it will throw if the receipt has a
+     * status that is not equal to 1. A status of 0 or null indicates that the
+     * transaction was mined, but failed. See:
+     * https://github.com/ethereum/wiki/wiki/JavaScript-API#web3ethgettransactionreceipt
+     * @param   txHash            Transaction hash
+     * @param   pollingIntervalMs How often (in ms) should we check if the transaction is mined.
+     * @param   timeoutMs         How long (in ms) to poll for transaction mined until aborting.
+     * @return  Transaction receipt with decoded log args.
+     */
+    public async awaitTransactionSuccessAsync(
+        txHash: string,
+        pollingIntervalMs: number = 1000,
+        timeoutMs?: number,
+    ): Promise<TransactionReceiptWithDecodedLogs> {
+        const receipt = await this.awaitTransactionMinedAsync(txHash, pollingIntervalMs, timeoutMs);
+        if (receipt.status !== 1) {
+            throw new Error('Transaction failed (transaction receipt did not have a status of 1)');
```

Comment:
Let's add the `txHash` in this error message to help the user while debugging.

Reply:
I'm not sure if the section in brackets is required?
`Transaction Failed: ${txHash}` feels like a good enough error message.